### PR TITLE
Added Fun. Tests for Add and View Dependents

### PR DESCRIPTION
**Pull request by leobcgov**
# Fixes or Implements [AB#9251](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/9251)

* [X] Enhancement
* [ ] Bug
* [ ] Documentation

## Description
- Added FUN test for Add Dependents.
- Updated tests for View Dependents and Covid Tab.
image

- Added tests re. menu.js
image

## Testing

* [X] Unit Tests Updated
* [ ] Functional Tests Updated
* [ ] Not Required

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes
NO

### Browsers Tested

* [X] Chrome
* [ ] Safari
* [ ] Edge
* [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant.  Include any specialized deployment steps here.  

</br>
</br>
</br>
___
  
## Reviewer Guidelines

A reviewer should examine the PR if the Status above indicates the PR is **Ready**.
Once ready, the reviewer should read the information below as well as supporting information like the Azure ticket to ensure they have a good understanding of the request.

The reviewer should then:

1. Is the PR template complete and accurate?

    Confirm by reading all sections below.

1. Confirm the merge branch

    A normal feature/bug branch should merge into the Dev branch.  Exceptions for hotfix and sprint merges do occur and the target would then be Master.

1. Do the code commits have an appropriate comment and include linkage to the azure ticket by including a reference to "AB#nnnn"?

1. Review the Code Changes

    1. Do the changes match the description?

        The code commits should reflect the documented description and nothing more.  If additional changes are included the reviewer should call these out and request that the developer document the changes in Azure and in the description or remove the changed code.

    1. useful tips

        * Any obvious errors or omissions?
        * Does the code match the style of other similar work?
        * Is the code easy to understand?
        * Is the code well documented?
        * DB Migrations are appropriate?

If the Answer to any of the questions above is No, then a request for change should be the result of the review.

### CODE REVIEW

File: Apps/WebClient/test/functional/e2e/cypress/integration/dependent/dependents.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            .then(phnNumber => expect(phnNumber).to.equal(maskedPHN));
+        cy.get('[data-testid=dependentDOB]')
+            .last().invoke('val')
+            .then(dateOfBirth => expect(dateOfBirth).to.equal(doB));
+        cy.get('[data-testid=dependentGender]')
+            .last().invoke('val')
+            .then(dateOfBirth => expect(dateOfBirth).to.equal(gendeFemale));        
+    })
+
+    it('Validate Covid Tab', () => {
```
Comment (mk08sh) [inline Apps/WebClient/test/functional/e2e/cypress/integration/dependent/dependents.js:115]: We should probably comment this out as it currently fails and will be tested on a different test

File: Apps/WebClient/test/functional/e2e/cypress/integration/dependent/dependents.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            .clear()
+            .blur()    
+            .should('have.class', 'is-invalid')
+        // Validate Last Name  
+        cy.get('[data-testid=lastNameInput]')
+            .should('be.enabled')
+            .clear()
+            .blur()
+            .should('have.class', 'is-invalid')
+        // Mandad this is broken and needs to be fixed to validate
```
Comment (sslaws) [inline Apps/WebClient/test/functional/e2e/cypress/integration/dependent/dependents.js:49]: What about this comment?

File: Apps/WebClient/test/functional/e2e/cypress/integration/dependent/dependents.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            .should('be.enabled', 'be.visible')
+            .click()
+        // Validate the modal is done, again should use modal id    
+        cy.get('[data-testid=newDependentModalText]').should('not.exist')
+    })
+
+    it('Validate Add', () => {
+        cy.get('[data-testid=addNewDependentBtn]')
+            .click();
+        //Validate Modal appears - note this should use a modal ID but we don't have one
```
Comment (sslaws) [inline Apps/WebClient/test/functional/e2e/cypress/integration/dependent/dependents.js:79]: We should be addressing this comment

**Review by sslaws — CHANGES_REQUESTED** [review]

File: Apps/WebClient/test/functional/e2e/cypress/integration/dependent/dependents.js (hunk 1 above)
Comment (leobcgov) [reply Apps/WebClient/test/functional/e2e/cypress/integration/dependent/dependents.js:115]: @mk08sh currently we can test if we can click on it and it shows "No Records" per screenshot above. I think we can add more tests to it once the covid issue is resolved

**Comment by leobcgov** [comment]
tests passed 
image

File: Apps/WebClient/test/functional/e2e/cypress/integration/dependent/dependents.js (hunk 3 above)
Comment (leobcgov) [reply Apps/WebClient/test/functional/e2e/cypress/integration/dependent/dependents.js]: addressed

File: Apps/WebClient/test/functional/e2e/cypress/integration/dependent/dependents.js (hunk 2 above)
Comment (leobcgov) [reply Apps/WebClient/test/functional/e2e/cypress/integration/dependent/dependents.js]: it should be good now.

File: Apps/WebClient/test/functional/e2e/cypress/integration/dependent/dependents.js (hunk 1 above)
Comment (leobcgov) [reply Apps/WebClient/test/functional/e2e/cypress/integration/dependent/dependents.js:114]: per our call, the tests are working fine now.

**Review by sslaws — APPROVED** [review]

**Review by FuriousLlama — APPROVED** [review]

**Pull request merged**